### PR TITLE
Add .out files to .gitignore, removed VS Code stuff and changed a few comments

### PR DESCRIPTION
**Pull request by GabrielRavier**
What this does :
- Adds *.out to .gitignore, filtering out out files (this was the old executable format for Linux)
- Removes VS Code from the .gitignore (this is the kind of stuff that should go in people's global gitignore)
- Changes a few comments

### CODE REVIEW

**Comment by Clownacy** [comment]
What generates .out files?

**Comment by GabrielRavier** [comment]
It was the executable format on Linux before COFF

**Comment by GabrielRavier** [comment]
I accidentally closed this by deleting the branch lol

**Comment by Clownacy** [comment]
This merge is broken right now.

**Comment by GabrielRavier** [comment]
Fixed it, also added a few more things (some comments, the VS Code stuff didn't go along with the standard .gitignore philosophy of "system-specific stuff goes in the global .gitignore")

File: .gitignore [hunk 1]
```diff
@@ -14,19 +14,18 @@ build_jp/*.rec
...
 build_en/CSE2
 build_en/CSE2d
 build_en/DoConfig
 build_en/DoConfigd
-
 build_jp/CSE2
 build_jp/CSE2d
 build_jp/DoConfig
 build_jp/DoConfigd
+*.out
```
Comment (Clownacy) [inline .gitignore]: Why not just in build_en and build_jp, like the exes?

Comment (GabrielRavier) [reply .gitignore]: Np fixing that

**Comment by GabrielRavier** [comment]
Fixed the global *.out issue

**Pull request merged**